codex/stores/repostore/store.nim

```
@@ -205,12 +205,28 @@ method delBlock*(self: RepoStore, cid: Cid): Future[?!void] {.async.} =
 
     if err =? (await self.updateQuotaUsage(minusUsed = res.released)).errorOption:
       return failure(err)
-  elif res.kind == InUse:
-    trace "Block in use, refCount > 0 and not expired"
-  else:
-    trace "Block not found in store"
 
-  return success()
+  success(res.kind)
+
+method delBlock*(self: RepoStore, cid: Cid): Future[?!void] {.async.} =
+  ## Delete a block from the blockstore when block refCount is 0 or block is expired
+  ##
+
+  logScope:
+    cid = cid
+
+  without outcome =? await self.delBlockInternal(cid), err:
+    return failure(err)
+
+  case outcome
+  of InUse:
```

why not return the actual error here, instead of just a string?

---

Can I return anything other than a string with `Future[?!void]`?

---

Oh, if you mean having a custom exception type with the error code - I've considered that, but didn't seem worth the trouble. We're not handling these errors on the client side, and it felt like adding this was just increasing complexity for no good reason.

---

Not all of the outcomes are errors, and the method returns `Future[?!void]`, so I don't have a lot of choices. I could create a custom exception type to describe the single case in which this is a failure, but didn't seem worth it, particularly since we're not handling these errors on the client side other than logging the error message anyway.

---

Ok, I think we can rework this later on.